tests/Feature_v2/Photo/PhotoEditTest.php

```diff
@@ -0,0 +1,236 @@
+{
+	public function testEditPhotoUnauthorizedForbidden(): void
+	{
+		$response = $this->patchJson('Photo', []);
+		$this->assertUnprocessable($response);
+
+		$response = $this->patchJson('Photo', [
+			'photo_id' => $this->photo1->id,
+			'title' => 'new title',
+			'description' => 'new description',
+			'tags' => ['tag1'],
+			'license' => 'none',
+			'upload_date' => '2021-01-01',
+		]);
+		$this->assertUnauthorized($response);
+
+		$response = $this->actingAs($this->userNoUpload)->patchJson('Photo::rename', [
+			'photo_id' => $this->photo1->id,
+			'title' => 'new title',
+			'description' => 'new description',
+			'tags' => ['tag1'],
+			'license' => 'none',
+			'upload_date' => '2021-01-01',
+		]);
+		$this->assertForbidden($response);
+	}
+
+	public function testEditPhotoAuthorizedOwner(): void
+	{
+		$response = $this->actingAs($this->userMayUpload1)->patchJson('Photo', []);
+		$this->assertUnprocessable($response);
+
+		$response = $this->actingAs($this->userMayUpload1)->patchJson('Photo', [
+			'photo_id' => $this->photo1->id,
+			'title' => 'new title',
+			'description' => 'new description',
+			'tags' => ['tag1'],
+			'license' => 'none',
+			'upload_date' => '2021-01-01',
+		]);
+		$this->assertOk($response);
+
+		$response = $this->actingAs($this->userMayUpload1)->getJsonWithData('Album', ['album_id' => $this->album1->id]);
+		$this->assertOk($response);
+		$response->assertJson([
+			'config' => [],
+			'resource' => [
+				'photos' => [
+					[
+						'id' => $this->photo1->id,
+						'title' => 'new title',
+						'description' => 'new description',
+						'tags' => ['tag1'],
+						'license' => 'none',
+						'created_at' => '2021-01-01T00:00:00+00:00',
+					],
+				],
+			],
+		]);
+	}
+
+	public function testSetStarPhotoUnauthorizedForbidden(): void
+	{
+		$response = $this->postJson('Photo::star', []);
+		$this->assertUnprocessable($response);
+
+		$response = $this->postJson('Photo::star', [
+			'photo_ids' => [$this->photo1->id],
+			'is_starred' => true,
+		]);
+		$this->assertUnauthorized($response);
+
+		$response = $this->actingAs($this->userNoUpload)->postJson('Photo::star', [
+			'photo_ids' => [$this->photo1->id],
+			'is_starred' => true,
+		]);
+		$this->assertForbidden($response);
+	}
+
+	public function testSetStarPhotoAuthorizedOwner(): void
+	{
+		$response = $this->actingAs($this->userMayUpload1)->postJson('Photo::star', []);
+		$this->assertUnprocessable($response);
+
+		$response = $this->actingAs($this->userMayUpload1)->postJson('Photo::star', [
+			'photo_ids' => [$this->photo1->id],
+			'is_starred' => true,
+		]);
+		$this->assertNoContent($response);
+
+		$response = $this->actingAs($this->userMayUpload1)->getJsonWithData('Album', ['album_id' => $this->album1->id]);
+		$this->assertOk($response);
+		$response->assertJson([
+			'config' => [],
+			'resource' => [
+				'photos' => [
+					[
+						'id' => $this->photo1->id,
+						'is_starred' => true,
+					],
+				],
+			],
+		]);
+	}
```

Comment:
These seem to be covered in PhotoStarTest.php?

Reply:
Good spot. Duplicate tests also on Move & Copy.